bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -1051,34 +1051,92 @@ bool SDFFeatures::AddSdfCollision(
          ++submeshIdx)
     {
       auto s = mesh->SubMeshByIndex(submeshIdx).lock();
-      auto vertexCount = s->VertexCount();
-      auto indexCount = s->IndexCount();
-      btAlignedObjectArray<btVector3> convertedVerts;
-      convertedVerts.reserve(static_cast<int>(vertexCount));
-      for (unsigned int i = 0; i < vertexCount; i++)
+      bool meshCreated = false;
+      if (meshSdf->Optimization() ==
+          ::sdf::MeshOptimization::CONVEX_DECOMPOSITION ||
+          meshSdf->Optimization() ==
+          ::sdf::MeshOptimization::CONVEX_HULL)
       {
-        convertedVerts.push_back(btVector3(
-              static_cast<btScalar>(s->Vertex(i).X()) * scale[0],
-              static_cast<btScalar>(s->Vertex(i).Y()) * scale[1],
-              static_cast<btScalar>(s->Vertex(i).Z()) * scale[2]));
+        std::size_t maxConvexHulls = 16u;
+        if (meshSdf->Optimization() == ::sdf::MeshOptimization::CONVEX_HULL)
+        {
+          /// create 1 convex hull for the whole submesh
+          maxConvexHulls = 1u;
+        }
+        else if (meshSdf->ConvexDecomposition())
+        {
+          // limit max number of convex hulls to generate
+          maxConvexHulls = meshSdf->ConvexDecomposition()->MaxConvexHulls();
+        }
+        std::vector<common::SubMesh> decomposed =
+          std::move(meshManager.ConvexDecomposition(*s.get(), maxConvexHulls));
+
+        gzdbg << "Optimizing mesh using convex decomposition. " << std::endl;
```

Comment:
we may want to remove this, but it has been very helpful for testing

Reply:
shortened the debug msg to one line. [1f2aaa7](https://github.com/gazebosim/gz-physics/pull/606/commits/1f2aaa7a24756c51f56b5d03ad56958926d38630)

I can also remove this msg completely if preferred.